python/addBTV.py

```diff
@@ -299,6 +299,91 @@ def add_BTV(process, runOnMC=False, onlyAK4=False, onlyAK8=False, keepInputs=['D
                       doc="DeepCSV light btag discriminator",
                       precision=10),
     )
+
+    FatJetVars = cms.PSet(
+        uncorrpt=Var("?availableJECSets().size()>0?correctedJet('Uncorrected').pt():pt()",
+                     float,
+                     doc="Uncorrected pT",
+                     precision=10),
+        residual=Var("?availableJECSets().size()>0 ? pt()/correctedJet('L3Absolute').pt() : 1. ",
+                     float,
+                     doc="residual",
+                     precision=10),
+        jes=Var("?availableJECSets().size()>0 ? pt()/correctedJet('Uncorrected').pt() : 1.",
+                float,
+                doc="jes (jet substructure)",
+            precision=10),
+        CombIVF=Var("bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags')",
+                float,
+                doc="combinedIVF",
+                precision=10),
+        DeepCSVBDisc=Var("bDiscriminator('pfDeepCSVJetTags:probb')+bDiscriminator('pfDeepCSVJetTags:probbb')",
```

Comment:
[Implementation itself, technical details & storage]
For the b discriminator (the sum of the individual b, bb related outputs) something similar to the CvsL/CvsB discriminator approach could be done to avoid getting irregular -2 bins: https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/jetsAK4_Puppi_cff.py#L83 -> also add the ternary operator to test whether the sum is >=0 and assign either that sum or -1 finally.